pkg/operator/controller/names.go

```diff
@@ -18,6 +18,19 @@ const (
 	// ControllerDeploymentLabel identifies a deployment as an ingress controller
 	// deployment, and the value is the name of the owning ingress controller.
 	ControllerDeploymentLabel = "ingresscontroller.operator.openshift.io/deployment-ingresscontroller"
+
+	// ControllerDeploymentUsesHostNetworkLabel identifies a deployment as
+	// using the host network.  This label is used for anti-affinity to
+	// prevent colocation of ingress controller pod replicas that use the
+	// host network.
+	ControllerDeploymentUsesHostNetworkLabel = "ingresscontroller.operator.openshift.io/uses-hostnetwork"
+
+	// ControllerDeploymentNonceLabel identifies an ingress controller deployment's
+	// generation.  This label is used for affinity, to colocate replicas of
+	// different generations of the same ingress controller, and for
+	// anti-affinity, to prevent colocation of replicas of the same
+	// generation of the same ingress controller.
+	ControllerDeploymentNonceLabel = "ingresscontroller.operator.openshift.io/nonce"
```

Comment:
Instead of exposing the host network and generation details through labels, would it be possible for the operator to compute whether a given ingresscontroller is co-locatable and then generically label with something like `ingresscontroller.operator.openshift.io/colocation-disabled`?